### PR TITLE
grc/qtgui: various Python 3 fixes

### PR DESCRIPTION
**Pull request by hauk142**

### CODE REVIEW

File: grc/core/generator/top_block.py [hunk 1]
```diff
@@ -227,7 +228,7 @@ def make_port_sig(port):
                 key = port.key
 
             if not key.isdigit():
-                key = repr(key)
+                key = re.findall(r'\d+', key)[0]
```
Comment (skoslowski) [inline grc/core/generator/top_block.py]: @hauk142 what is the reasoning for this? it looks like your breaking message connections.

Comment (hauk142) [reply grc/core/generator/top_block.py]: Trying to run a generated flowgraph without this fix gives me a NotImplementedError. I've included the error message in the commit message :)

This fix takes for example 'in2' and changes it to 2.

Comment (skoslowski) [reply grc/core/generator/top_block.py]: Yes, that should have been in a WIP commit. Didn't realize that got merged. Checkout _init_ports() in block.py. Removing in and out there allows you to revert this change here

Comment (hauk142) [reply grc/core/generator/top_block.py]: Right! I'll have a look

**Pull request merged**